fairscale/nn/data_parallel/fully_sharded_data_parallel.py

```diff
@@ -204,6 +204,11 @@ class FullyShardedDataParallel(nn.Module):
             if ``True``, reshard parameters after the forward pass. This saves
             memory but slows training. This is only relevant when resharding
             individual layers.
+        disable_reshard_on_root (bool, Optional):
+            if ``True``, reshard_after_forward will be set to False if the module is a root.
+            For some cases, the full parameters of root do not need to be sharded in order to
+            improve performance. It is because the parameters are needed immediately
+            for the backward pass.
```

Comment:
Can you add a sentence on why do we want to disable this in some cases?

Reply:
I have updated my description to further explain the reason. 
Please tell me if there is still other problem. Thanks.
By the way, the default value **disable_reshard_on_root** is True so that my update will not change the original behavior of reshard_after_forward